pkg/cmd/dictionary/entries/browse/browse.go

```diff
@@ -0,0 +1,161 @@
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			return shared.DictionaryNames(), cobra.ShellCompDirectiveNoFileComp
+		},
+		Short: "Browse dictionary entries",
+		Long: heredoc.Docf(`
+			This command retrieves all entries from the specified %s dictionnaries.
+		`, cs.Bold("custom")),
+		Example: heredoc.Doc(`
+			# Retrieve all entries from the "stopword" dictionary (doesn't include default stopwords)
+			$ algolia dictionary entries browse stopword
+
+			# Retrieve all entries from the "stopword" and "plural" dictionnaries
+			$ algolia dictionary entries browse stopword plural
+
+			# Retrieve all entries from all dictionnaries
+			$ algolia dictionary entries browse --all
+
+			# Retrieve all entries from the "stopword" dictionnaries (including default stopwords)
+			$ algolia dictionary entries browse stopword --include-defaults
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if opts.All && len(args) > 0 || !opts.All && len(args) == 0 {
+				return cmdutil.FlagErrorf("Either specify dictionaries' names or use --all to browse all dictionaries")
+			}
+
+			if opts.All {
+				opts.Dictionnaries = []search.DictionaryName{search.Stopwords, search.Plurals, search.Compounds}
+			} else {
+				opts.Dictionnaries = make([]search.DictionaryName, len(args))
+				for i, dictionnary := range args {
+					opts.Dictionnaries[i] = search.DictionaryName(dictionnary)
+				}
+			}
+
+			return runBrowseCmd(opts)
+		},
+	}
+
+	cmd.Flags().BoolVarP(&opts.All, "all", "a", false, "browse all dictionnaries")
+	cmd.Flags().BoolVarP(&opts.IncludeDefaultStopwords, "include-defaults", "d", false, "browse dictionaries and include default stopwords")
+
+	opts.PrintFlags.AddFlags(cmd)
+
+	return cmd
+}
+
+// runBrowseCmd executes the browse command
+func runBrowseCmd(opts *BrowseOptions) error {
+	cs := opts.IO.ColorScheme()
+	client, err := opts.SearchClient()
+	if err != nil {
+		return err
+	}
+
+	dictionaries := opts.Dictionnaries
+
+	p, err := opts.PrintFlags.ToPrinter()
+	if err != nil {
+		return err
+	}
+
+	for _, dictionnary := range dictionaries {
+		pageCount := 0
+		maxPages := 1
+
+		// implement infinite pagination
+		for pageCount < maxPages {
+			res, err := client.SearchDictionaryEntries(dictionnary, "", opt.HitsPerPage(1000), opt.Page(pageCount))
+			if err != nil {
+				return err
+			}
+
+			maxPages = res.NbPages
+
+			data, err := json.Marshal(res.Hits)
+			if err != nil {
+				return fmt.Errorf("cannot unmarshal dictionary entries: error while marshalling original dictionary entries: %v", err)
+			}
+
+			var entries []DictionaryEntry
+			err = json.Unmarshal(data, &entries)
+			if err != nil {
+				return fmt.Errorf("cannot unmarshal dictionary entries: error while unmarshalling original dictionary entries: %v", err)
+			}
+
+			if len(entries) == 0 {
+				fmt.Fprintf(opts.IO.Out, "%s No entries in %s dictionary.\n\n", cs.WarningIcon(), dictionnary)
```

Comment:
Not 100% sure we want to display a warning here.
Explanation: Potentially, this command could be used to copy one or multiple dictionaries entries from one application to another. If this statement is printed in the middle of an export / import pipe, it will break.
What I would do is only printing this statement if ALL the dictionary are empty (so we don't care if the copy is broken because there is nothing to copy anyway).
WDYT?